lib/dal/src/attribute/value/dependent_value_graph.rs

```diff
@@ -51,8 +52,19 @@ impl DependentValueGraph {
         > = HashMap::new();
         let mut dependent_value_graph = Self::new();
 
+        let workspace_snapshot = ctx.workspace_snapshot()?;
+
         let mut work_queue = VecDeque::from_iter(values);
         while let Some(current_attribute_value_id) = work_queue.pop_front() {
+            if workspace_snapshot
+                .try_get_node_index_by_id(current_attribute_value_id)
+                .await?
+                .is_none()
+            {
+                warn!("Attribute Value {current_attribute_value_id} missing, skipping it in DependentValueGraph");
+                continue;
+            }
+
```

Comment:
Please add a comment here along the lines of
```
// It's possible that one or more of the initial AttributeValueIds provided by the enqueued DependentValuesUpdate job
// may have been removed from the snapshot between when the DVU job was created and when we're processing
// things now. This could happen if there are other modifications to the snapshot before the DVU job starts
// executing, as the job always operates on the current state of the change set's snapshot, not the state at the time
// the job was created.
```

I don't think the warning is necessary. It could be dropped down to `debug`.

Reply:
Done